src/component/contents/DraftEditorContents.react.js

```diff
@@ -192,26 +194,88 @@ class DraftEditorContents extends React.Component {
         <Component {...componentProps} />,
       );
 
+      // wrapperTemplate is the internal DraftJS wrapping template to wrap various blocks.
+      // customWrapperTemplate is a user provided component 
+      // and will also wrap blocks on top of wrapping by wrapperTemplate.
       processedBlocks.push({
         block: child,
-        wrapperTemplate,
         key,
         offsetKey,
+        wrapperTemplate,
+        customWrapperTemplate,
       });
 
-      if (wrapperTemplate) {
+      if (wrapperTemplate || customWrapperTemplate) {
         currentDepth = block.getDepth();
       } else {
         currentDepth = null;
       }
       lastWrapperTemplate = wrapperTemplate;
+      lastCustomWrapperTemplate = customWrapperTemplate;
     }
 
-    // Group contiguous runs of blocks that have the same wrapperTemplate
+    // Group contiguous runs of blocks
     const outputBlocks = [];
-    for (let ii = 0; ii < processedBlocks.length; ) {
+    for (let ii = 0; ii < processedBlocks.length;) {
       const info = processedBlocks[ii];
```

Comment:
Do you think it makes sense to do something like this instead of storing `info`?
```javascript
let internalWrapperTemplate = null;
let customWrapperTemplate = null;
```
It's starting to get confusing between `info` and `infoBlock`.

Reply:
It needs more data from the reference block unfortunately (offsetKey + key)
I renamed it `blockRef` which is hopefully an improvement. If we're willing to reuse the offsetKey + key, I can simplify it. Code below:

```
        const customWrapperElement = React.cloneElement(
          info.customWrapperTemplate,
          {
            key: info.key + '-custom-wrap',
            'data-offset-key': info.offsetKey,
          },
          blocksInInternalWrapper,
        );
        outputBlocks.push(customWrapperElement);
```